### PR TITLE
Create HTTP classes

### PR DESCRIPTION
**Pull request by ahesselgrave**
Create Request, Response, and appropriate helpers and tests.

### CODE REVIEW

File: src/http/constants.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#ifndef STATUS_H
```
Comment (jobu95) [inline src/http/constants.h]: For consistency shouldn't this be CONSTANTS_H?

File: src/http/constants.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	const int HTTP_200_OK = 200;
+	const int HTTP_404_NOT_FOUND = 404;
+
+	inline std::string getReasonPhrase(int fromStatus) {
+		switch(fromStatus) {
+			case HTTP_200_OK: return "OK";
+			case HTTP_404_NOT_FOUND: return "NOT FOUND";
+			default: 
+				std::cerr << "invalid status code" << std::endl;
+				exit(-1);
```
Comment (jobu95) [inline src/http/constants.h]: I've been exiting with status 1. Do you want to switch everything to status -1?

File: src/http/message.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+/*
+ * Interface for HTTP messages. Used to hold common code
+ * for HTTP Requests and Responses
+ */
+class Message {
+protected:
+	// Effectively namespaced constant for carriage returns
+
+	std::vector<std::string> headers_;
```
Comment (jobu95) [inline src/http/message.h]: I haven't been using trailing underscores for member variables. Do you want to from now on?

**Review by jobu95 — CHANGES_REQUESTED** [review]
I asked for a couple minor changes, but if you decide not to do them feel free to merge.

File: src/http/constants.h (hunk 1 above)
Comment (ahesselgrave) [reply src/http/constants.h]: woops yeah, I renamed the file but forgot to change the header guards, I'll fix

File: src/http/constants.h (hunk 2 above)
Comment (ahesselgrave) [reply src/http/constants.h]: Technically we should have more defined error codes, but I'll change them to 1 for consistency as a general error and we can specialize error codes later if we need it

Comment (jobu95) [reply src/http/constants.h]: Yeah I agree. Thanks.

File: src/http/message.h (hunk 3 above)
Comment (ahesselgrave) [reply src/http/message.h]: I'm following the naming convention [https://google.github.io/styleguide/cppguide.html#Variable_Names](here) cause I saw it in lecture lol. I'm not attached to anything but I would prefer consistency.

Comment (jobu95) [reply src/http/message.h]: Ok, I'll add trailing underscores to my member variables.

File: src/http/constants.h (hunk 2 above)
Comment (ahesselgrave) [reply src/http/constants.h]: wait actually wtf I changed that to throw an exception. I guess I forgot to update the commits.

**Comment by ahesselgrave** [comment]
I'm gonna assume this is ok since I changed what you requested and it passed. Gotta move forward.

**Pull request merged**